apps/app_audiosocket.c

```diff
@@ -183,39 +173,56 @@ static int audiosocket_run(struct ast_channel *chan, const char *id, int svc)
 	chanName = ast_channel_name(chan);
 
 	while (1) {
-		ms = -1;
+		/* Timeout is hard-coded currently, could be made into an
+		   argument if needed, but 2 seconds seems like a realistic
+		   time range to give. */
 		targetChan = ast_waitfor_nandfds(&chan, 1, &svc, 1, NULL, &outfd, &ms);
+		ms = MAX_WAIT_TIMEOUT_MSEC;
+
 		if (targetChan) {
+			/* Receive audio frame from connected channel. */
 			f = ast_read(chan);
 			if (!f) {
+				ast_log(LOG_WARNING, "Failed to receive frame from channel '%s' connected to AudioSocket server '%s'", chanName, server);
 				return -1;
 			}
 
-			if (f->frametype == AST_FRAME_VOICE) {
+			if (f->frametype == AST_FRAME_VOICE || f->frametype == AST_FRAME_DTMF) {
 				/* Send audio frame to audiosocket */
```

Comment:
This is no longer accurate, and now this is supporting DTMF but not documented elsewhere as such

Reply:
hello @jcolp,

could you please elaborate on "this is no longer accurate"? 

the app used to filter on AST_FRAME_VOICE frames, I just added DTMF frames. The app description has been updated to specify DTMF frames are forwarded too.

Reply:
The comment says "Send audio frame to audiosocket"